Find a functionally equivalent code:android.transition.Transition.end:COMMENT
Method Modifier: protected   hidden      
Comment:/**
 * This method is called automatically by the Transition and
 * TransitionSet classes when a transition finishes, either because
 * a transition did nothing (returned a null Animator from
 * {@link Transition#createAnimator(ViewGroup, TransitionValues,
 * TransitionValues)}) or because the transition returned a valid
 * Animator and end() was called in the onAnimationEnd()
 * callback of the AnimatorListener.
 *
 * @hide
 */

@@ -9,25 +9,23 @@
             }
         }
         for (int i = 0; i < mStartValues.itemIdValues.size(); ++i) {
-            TransitionValues tv = mStartValues.itemIdValues.valueAt(i);
-            View v = tv.view;
-            if (v.hasTransientState()) {
-                v.setHasTransientState(false);
+            View view = mStartValues.itemIdValues.valueAt(i);
+            if (view != null) {
+                view.setHasTransientState(false);
             }
         }
         for (int i = 0; i < mEndValues.itemIdValues.size(); ++i) {
-            TransitionValues tv = mEndValues.itemIdValues.valueAt(i);
-            View v = tv.view;
-            if (v.hasTransientState()) {
-                v.setHasTransientState(false);
+            View view = mEndValues.itemIdValues.valueAt(i);
+            if (view != null) {
+                view.setHasTransientState(false);
             }
         }
         mEnded = true;

